app/android/src/uk/co/lutraconsulting/CameraActivity.java

```diff
@@ -83,75 +85,91 @@ private File createImageFile(String targetPath) throws IOException {
         String currentPhotoPath;
         String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());
         String imageFileName = "JPEG_" + timeStamp + "_";
-        
-        cameraFile = File.createTempFile(imageFileName,  /* prefix */
-                                         ".jpg",         /* suffix */
-                                         getCacheDir()      /* directory */
-                                         );
+
+        cameraFile = File.createTempFile(
+                imageFileName, /* prefix */
+                ".jpg", /* suffix */
+                getCacheDir() /* directory */
+        );
 
         // Save a file: path for use with ACTION_VIEW intents
         currentPhotoPath = cameraFile.getAbsolutePath();
-        Log.d(TAG, "currentPhotoPath: "+currentPhotoPath);
+        Log.d(TAG, "currentPhotoPath: " + currentPhotoPath);
         return cameraFile;
     }
-    
+
     @Override
     protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         Log.d(TAG, "onActivityResult()");
-        Log.d(TAG, "request: "+requestCode);
-        Log.d(TAG, "resultCode: "+resultCode);
+        Log.d(TAG, "request: " + requestCode);
+        Log.d(TAG, "resultCode: " + resultCode);
+        orientationSensor.Unregister();
+
         if (requestCode == CAMERA_CODE && resultCode == Activity.RESULT_OK) {
+            Log.d(TAG, "tmp exists: " + cameraFile.exists());
+            Log.d(TAG, "tmp path: " + cameraFile.getAbsolutePath());
 
-            Log.d(TAG, "tmp exists: "+cameraFile.exists());
-            Log.d(TAG, "tmp path: "+cameraFile.getAbsolutePath());
-            try{
+            try {
+                extendGPSExifData(cameraFile.lastModified());
                 copyFile(cameraFile, new File(targetPath, cameraFile.getName()));
                 if (data == null) {
-                    data = getIntent();                
+                    data = getIntent();
                 }
                 data.putExtra("__RESULT__", cameraFile.getAbsolutePath());
-                setResult(Activity.RESULT_OK, data);   
-                
-            }catch(IOException e){
+                setResult(Activity.RESULT_OK, data);
+
+            } catch (IOException e) {
                 Intent intent = this.getIntent();
                 if (data == null) {
-                    data = getIntent();                
+                    data = getIntent();
                 }
                 data.putExtra("__RESULT__", e.getMessage());
-                setResult(Activity.RESULT_CANCELED, data);   
+                setResult(Activity.RESULT_CANCELED, data);
             }
-            
+
             // TODO: after copy, verify if is correctly copied and then remove the old one
-            
         }
         finish();
     }
 
+    private void extendGPSExifData(long captureTime) {
+        int bearing = getValueByTime(orientationSensor.m_azimuth_data, captureTime);
```

Comment:
what if the sensors are not available for some reason - we should detect it here and not write any direction info as it may be confusing